.travis.yml

```diff
@@ -4,47 +4,138 @@ env:
   global:
     - NUMPY=numpy
     - MPL=matplotlib
+    - PYPROJ=pyproj
     - TEST_ARGS=--no-pep8
+    - BUILD_LIBGEOS=false
+
+# install apt binary packages
+addons:
+  apt:
+    packages:
+      - libgeos-3.3.8   
+      # matplotlib requirements
+      - libpng12-dev
+      - libfreetype6-dev
+      # helpers utilities
+      - autotools-dev
+      - wget
 
 language: python
 
+cache: pip
+
+
+# matrix generates 6 test cases
 matrix:
   include:
+    # "minimum requirements" environments
+    # test the bare minimum versions 
+    # compiles internal libgeos
+    # Notes:
+    #   numpy 1.7.0 is the first version that works out of the box for Python 2.6 and 3.3
+    #   numpy 1.5.1 is the earliest version to compile and *might* be able to work on Python 2.6,
+    #         if two unit tests are skipped due to numpy.copy(a,order) not having 'order' parameter
+    #         1.6.0-1.6.2 works out of the box for Python 2.6
+    #   matplotlib 1.2.0 was found to be the earliest version that installs
     - python: 2.6
-      env: NUMPY=numpy==1.6
-    - python: 2.7
+      env: 
+        - NUMPY=numpy==1.7.0
+        - MPL=matplotlib==1.2.0
+        - BUILD_LIBGEOS=internal
     - python: 3.3
+      env: 
+        - NUMPY=numpy==1.7.0
+        - MPL=matplotlib==1.2.0
+        - BUILD_LIBGEOS=internal
+
+    # "middle of the road" environment
+    # use prepackaged binaries
+    # if no prepackaged binary available, use a previous stable version
     - python: 3.4
+      env: 
+        - MPL=matplotlib==1.4.3
+#        - NUMPY="numpy>=1.9.0,<1.10.0"  # use the default install of numpy
+      
+    # "latest and greatest" stable environments
+    # this uses the current stable versions
+    # pip installs latest stable versions automatically
+    - python: 2.7
+      env:
+        - NUMPY=numpy>=1.10.0
+        - MPL=matplotlib>=1.5.0
+        - BUILD_LIBGEOS=3.5.0
```

Comment:
Rather than hardcoding the latest version which will for sure fall out off sync I would suggest these envs to pass `--upgrade` to pip that way we are sure that the latest version is installed.